tool/tctl/common/plugin/entraid.go

```diff
@@ -0,0 +1,401 @@
+			}
+			if _, err = args.authClient.CreateIntegration(ctx, integrationSpec); err != nil {
+				return trace.Wrap(err, "failed to create Azure OIDC integration")
+			}
+		}
+	}
+
+	credentialsSource := types.EntraIDCredentialsSource_ENTRAID_CREDENTIALS_SOURCE_OIDC
+	if inputs.entraID.useSystemCredentials {
+		credentialsSource = types.EntraIDCredentialsSource_ENTRAID_CREDENTIALS_SOURCE_SYSTEM_CREDENTIALS
+	}
+	req := &pluginspb.CreatePluginRequest{
+		Plugin: &types.PluginV1{
+			Metadata: types.Metadata{
+				Name: inputs.name,
+				Labels: map[string]string{
+					"teleport.dev/hosted-plugin": "true",
+				},
+			},
+			Spec: types.PluginSpecV1{
+				Settings: &types.PluginSpecV1_EntraId{
+					EntraId: &types.PluginEntraIDSettings{
+						SyncSettings: &types.PluginEntraIDSyncSettings{
+							DefaultOwners:     inputs.entraID.defaultOwners,
+							SsoConnectorId:    inputs.entraID.authConnectorName,
+							CredentialsSource: credentialsSource,
+							TenantId:          settings.tenantID,
+						},
+						AccessGraphSettings: tagSyncSettings,
+					},
+				},
+			},
+		},
+	}
+
+	_, err = args.plugins.CreatePlugin(ctx, req)
+	if err != nil {
+		if !trace.IsAlreadyExists(err) || !inputs.entraID.force {
+			return trace.Wrap(err)
+		}
+		if _, err = args.plugins.DeletePlugin(ctx, &pluginspb.DeletePluginRequest{
+			Name: inputs.name,
+		}); err != nil {
+			return trace.Wrap(err)
+		}
+		if _, err = args.plugins.CreatePlugin(ctx, req); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	fmt.Printf("Successfully created EntraID plugin %q\n\n", p.install.name)
+
+	return nil
+}
+
+func buildScript(proxyPublicAddr string, authConnectorName string, accessGraph, skipOIDCSetup bool) (string, error) {
+	// The script must execute the following command:
+	argsList := []string{
+		"integration", "configure", "azure-oidc",
+		fmt.Sprintf("--proxy-public-addr=%s", shsprintf.EscapeDefaultContext(proxyPublicAddr)),
+		fmt.Sprintf("--auth-connector-name=%s", shsprintf.EscapeDefaultContext(authConnectorName)),
+	}
+
+	if accessGraph {
+		argsList = append(argsList, "--access-graph")
+	}
+
+	if skipOIDCSetup {
+		argsList = append(argsList, "--skip-oidc-integration")
+	}
+
+	script, err := oneoff.BuildScript(oneoff.OneOffScriptParams{
+		TeleportArgs:   strings.Join(argsList, " "),
+		SuccessMessage: "Success! You can now go back to the Teleport Web UI to use the integration with Azure.",
+	})
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	return script, nil
+}
+
+func getProxyPublicAddr(ctx context.Context, authClient authClient) (string, error) {
+	pingResp, err := authClient.Ping(ctx)
+	if err != nil {
+		return "", trace.Wrap(err, "failed fetching cluster info")
+	}
+	proxyPublicAddr := pingResp.GetProxyPublicAddr()
+	oidcIssuer, err := oidc.IssuerFromPublicAddress(proxyPublicAddr, "")
+	return oidcIssuer, trace.Wrap(err)
+}
+
+func readTAGCache(fileLoc string) (*azureoidc.TAGInfoCache, error) {
```

Comment:
Nit: Add a godoc on this method explaining what TAG cache is and why we might want to read it.